### PR TITLE
feat(demos): Add theme switcher to theme demo page

### PR DESCRIPTION
**Pull request by acdvorak**
This PR also adds hot reloading of stylesheets that have a `data-hot` attribute.

Currently, only the theme demo page has a `<link rel="stylesheet" href="..." data-hot>` element, but I plan to create follow-up PRs to add hot swapping to the remaining demo pages. Hot swapping was broken when we switched to CSS files in #1916.

When Webpack starts recompiling our Sass, the theme demo page displays a linear-progress bar in the toolbar until compilation finishes, at which point it automatically swaps in the new stylesheet.

I tested this PR in:

- Chrome 63 (desktop and mobile)
- Firefox 57 (desktop)
- Safari 11 (desktop and mobile)
- IE 11
- Edge 14

### CODE REVIEW

File: demos/theme/index.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      if (evt.target.classList.contains(classes.COMPONENT_SECTION_PERMALINK)) {
+        evt.target.parentElement.classList.remove(classes.COMPONENT_SECTION_HEADING_FOCUS_WITHIN);
+      }
+    });
+
+    this.window_.addEventListener('hashchange', () => this.flashCurrentSection_());
+    this.flashCurrentSection_();
+  }
+
+  // TODO(acdvorak): Replace page URL hash with current section on scroll
```
Comment (kfranqueiro) [inline demos/theme/index.js]: Reminder that this TODO is here, whether or not you want to deal with it right now; given that we have links in each heading I don't think it's critical.

File: demos/theme/_theme-shared.scss [hunk 2]
```diff
@@ -61,10 +66,35 @@ body {
   background-color: $mdc-theme-background;
 }
 
+figure {
```
Comment (kfranqueiro) [inline demos/theme/_theme-shared.scss]: This seems to be redundant of styles 10 lines above this. Remove this or remove those?

File: demos/theme/index.js (hunk 1 above)
Comment (acdvorak) [reply demos/theme/index.js]: Removed TODO. Thanks!

File: demos/theme/_theme-shared.scss (hunk 2 above)
Comment (acdvorak) [reply demos/theme/_theme-shared.scss]: Done

File: demos/interactivity.js [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+  initialize() {}
+
+  /**
+   * @param {string} id
+   * @param {!Document|!Element=} opt_root
+   * @return {?Element}
+   * @protected
+   */
+  getElementById_(id, opt_root) {
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply demos/interactivity.js]: Done

File: demos/dom.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * @param {string} query
+ * @param {!Document|!Element=} opt_root
+ * @return {!Array<!Element>}
+ */
+export function getAll(query, opt_root) {
+  opt_root = opt_root || document;
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply demos/dom.js]: Done

**Pull request merged**